cmd/soroban-cli/src/signer/secure_store_entry.rs

```diff
@@ -0,0 +1,131 @@
+use stellar_strkey::ed25519::PublicKey;
+
+use crate::{
+    print::Print,
+    xdr::{self, DecoratedSignature},
+};
+
+#[cfg(feature = "additional-libs")]
+use crate::{
+    signer::keyring::{self, StellarEntry},
+    xdr::{Signature, SignatureHint},
+};
+#[cfg(feature = "additional-libs")]
+use std::sync::Arc;
+
+use ed25519_dalek::Signature as Ed25519Signature;
+
+use sep5::SeedPhrase;
+
+#[cfg(feature = "additional-libs")]
+const ENTRY_SERVICE: &str = "org.stellar.cli";
+pub(crate) const ENTRY_PREFIX: &str = "secure_store:";
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[cfg(feature = "additional-libs")]
+    #[error(transparent)]
+    Keyring(#[from] keyring::Error),
+
+    #[error(transparent)]
+    TryFromSlice(#[from] std::array::TryFromSliceError),
+
+    #[error(transparent)]
+    Xdr(#[from] xdr::Error),
+
+    #[error("Secure Store keys are not allowed: additional-libs feature must be enabled")]
+    FeatureNotEnabled,
+}
+
+#[derive(Debug, Clone)]
+pub struct SecureStoreEntry {
+    pub hd_path: Option<usize>,
+    #[cfg(feature = "additional-libs")]
+    pub entry: Arc<StellarEntry>,
```

Comment:
Since this is wrapped in an Arc we are able to clone the same `StellarEntry` to reuse in the whole command